web/src/engine/websites/ManhwaHentaiMe.ts

```diff
@@ -1,21 +1,70 @@
 
-@Madara.MangaCSS(/^{origin}\/webtoon\/[^/]+\/$/, 'div.post-title h1')
-@Madara.MangasMultiPageAJAX()
-@Madara.ChaptersSinglePageAJAXv2()
+type MangaID = {
+     post: string;
+     slug: string;
+}
+
+function MangaInfoExtractor(anchor: HTMLAnchorElement) {
+    const container = anchor.closest<HTMLElement>('div.page-item-detail, div.manga');
+    const post = container?.querySelector<HTMLElement>('div[id*="manga-item-"]')?.getAttribute('id').match(/(\d+$)/)[1] || '';
+    const slug = anchor.pathname;
+    const title = anchor.text.replace(/Webtoon Manhwa Hentai$/, '').trim();
+    const id = JSON.stringify({ post, slug });
+    return { id, title };
+}
+
+@Common.MangasMultiPageCSS('/home/page/{page}/', 'div.post-title h3 a, div.post-title h5 a', 1, 1, 0, MangaInfoExtractor)
 @Madara.PagesSinglePageCSS()
 @Common.ImageAjax()
 export default class extends DecoratableMangaScraper {
 
-    public constructor() {
-        super('manhwahentaime', 'ManhwaHentai.me', 'https://manhwahentai.me', Tags.Media.Manhua, Tags.Media.Manhwa, Tags.Language.English);
+    public constructor(id: string = 'manhwahentaime', label: string = 'ManhwaHentai.me', url: string = 'https://manhwahentai.me', tags: Tag[] = [Tags.Media.Manhua, Tags.Media.Manhwa, Tags.Language.English]) {
+        super(id, label, url, ...tags);
     }
 
     public override get Icon() {
         return icon;
     }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return new RegExpSafe(`^${this.URI.origin}/webtoon/[^/]+\/$`).test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const uri = new URL(url);
+        const data = await FetchHTML(new Request(uri));
+        const post = data.querySelector<HTMLElement>('div#star[data-id]')?.dataset?.id;
+        const slug = uri.pathname;
+        const element = data.querySelector<HTMLElement>('div.post-title h1');
+        const title = Common.AnchorInfoExtractor(false, 'span').call(this, element).title.replace(/Webtoon Manhwa Hentai$/, '').trim();
+        return new Manga(this, provider, JSON.stringify({ post, slug }), title);
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const id = JSON.parse(manga.Identifier) as MangaID;
+        const request = new Request(new URL('/wp-admin/admin-ajax.php', this.URI), {
+            method: 'POST',
+            body: new URLSearchParams({
+                action: 'ajax_chap',
+                post_id: id.post
+            }).toString(),
+            headers: {
+                'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
+                Referer: this.URI.href
+            }
+        });
+
+        const data = await FetchCSS(request, 'ul li.wp-manga-chapter > a');
+        return data.map(element => {
+            const { id, title } = Common.AnchorInfoExtractor(false, 'span').call(this, element);
```

Comment:
Same extractor is created and used in other function as well, consider to extract for re-usage.